tachyon/zk/plonk/vanishing/graph_evaluator_unittest.cc

```diff
@@ -126,6 +126,363 @@ TEST_F(GraphEvaluatorTest, Challenges) {
             CalculationInfo(Calculation::Store(ValueSource::Challenge(1)), 0));
 }
 
-// TODO(chokobole): AddTest for Negated, Sum, Product and Scale.
+TEST_F(GraphEvaluatorTest, ConstantNegated) {
+  GraphEvaluator<GF7> graph_evaluator;
+  for (size_t i = 0; i < 7; ++i) {
+    std::unique_ptr<Expression<GF7>> expr = ExpressionFactory<GF7>::Negated(
+        ExpressionFactory<GF7>::Constant(GF7(i)));
+    ValueSource source = graph_evaluator.Evaluate(expr.get());
+    if (i == 0) {
+      EXPECT_TRUE(source.IsZeroConstant());
+    } else if (i == 6) {
+      // Negation of 6 in GF7 is 1.
+      EXPECT_EQ(source, ValueSource::OneConstant());
+    } else if (i == 5) {
+      // Negation of 5 in GF7 is 2.
+      EXPECT_EQ(source, ValueSource::TwoConstant());
+    } else {
+      // Other constants besides 0, 1, and 2 are stored sequentially regardless
+      // of their value.
+      EXPECT_EQ(source, ValueSource::Constant(i + 2));
+    }
+  }
+}
+
+TEST_F(GraphEvaluatorTest, Negated) {
+  std::vector<ColumnInfo> test_column_infos = GenerateTestColumnInfos();
+  // clang-format off
+  // 1. test_column₀
+  //      {column_index: 0, rotation: 1, rotation_index: 0, calculation_index: 0}
+  //      intermediate = {test_column₀, Negate(test_column₀)}
```

Comment:
Can you fix all intermediate/intermediates to calculations_? Maybe renaming `calculations_` in `GraphEvaluator` to `intermediates_` might be another great choice as well.